master/internal/workspace/postgres_workspace.go

```diff
@@ -17,6 +20,43 @@ func WorkspaceByName(ctx context.Context, workspaceName string) (*model.Workspac
 	return &w, nil
 }
 
+// WorkspaceIDsFromNames returns an unordered slice of workspaceIDs that correlate with the given
+// workspace names.
+func WorkspaceIDsFromNames(ctx context.Context, workspaceNames []string) (
+	[]int32, error,
+) {
+	var workspaces []model.Workspace
+	err := db.Bun().NewSelect().
+		Model(&workspaces).
+		Where("name IN (?)", bun.In(workspaceNames)).
+		Scan(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if len(workspaces) != len(workspaceNames) {
+		workspaceNamesMap := map[string]bool{}
+		for _, name := range workspaceNames {
+			workspaceNamesMap[name] = false
+		}
+		for _, workspace := range workspaces {
+			workspaceNamesMap[workspace.Name] = true
+		}
+		missing := make([]string, len(workspaceNames)-len(workspaces))
+		for name, found := range workspaceNamesMap {
+			if !found {
+				missing = append(missing, name)
+			}
+		}
```

Comment:
Nitpick, if we want to use two iterations:
1. Get a set of valid workspace names from DB.
2. Iterate over `workspaceNames`. If a `workspaceName` doesn't exist in the set, add it to the `missing` slice.